rasa/core/policies/sklearn_policy.py

```diff
@@ -163,9 +163,10 @@ def _preprocess_data(self, X: Data) -> np.ndarray:
             )
 
         attribute_data = {
-            attribute: self._get_features_for_attribute(X[attribute], attribute)
-            for attribute in X
+            attribute: self._get_features_for_attribute(X[attribute]) for attribute in X
         }
+        # turning it into OrderedDict so that the order of features is the same
+        attribute_data = OrderedDict(attribute_data)
         attribute_data = [features for key, features in attribute_data.items()]
```

Comment:
@evgeniiaraz can it be simplified:
```
features = [self._get_features_for_attribute(attribute_data, attribute) for attribute, attribute_data in data.items()]
return np.concatenate(features, axis=-1)
```

Reply:
aaah, we need to order them

Reply:
we could turn `X` into `OrderedDict` and then do the simplification as you suggested